internal/docker/docker.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 deploy-agent authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+const (
+	defaultEndpoint         = "unix:///var/run/docker.sock"
+	streamInactivityTimeout = time.Minute
+)
+
+type AuthConfig docker.AuthConfiguration
+
+type Container struct {
+	ID string
+}
+
+type Image struct {
+	ID         string
+	registry   string
+	repository string
+	tag        string
+}
+
+func (i Image) Name() string {
+	return i.registry + "/" + i.repository
+}
+
+func (i Image) String() string {
+	return i.registry + "/" + i.repository + ":" + i.tag
+}
+
+type Client struct {
+	api *docker.Client
+}
+
+func NewClient(endpoint string) (*Client, error) {
+	if endpoint == "" {
+		endpoint = defaultEndpoint
+	}
+	cli, err := docker.NewClient(endpoint)
```

Comment:
I think it would be safer for us to configure the docker client timeout and override its dialer to add a dial timeout as well, something similar to what we do in BS: https://github.com/tsuru/bs/blob/b3ed6908d02c8316a237ddc16267108ed1577829/container/container.go#L75-L77